src/Microsoft.OData.Core/ODataMetadataOutputContext.cs

```diff
@@ -267,7 +268,13 @@ protected override async ValueTask DisposeAsyncCore()
         private void WriteMetadataDocumentImplementation()
         {
             IEnumerable<EdmError> errors;
-            if (!CsdlWriter.TryWriteCsdl(this.Model, this.xmlWriter, CsdlTarget.OData, out errors))
+
+            CsdlXmlWriterSettings settings = new CsdlXmlWriterSettings
+            {
+                UseLegacyVariableCasing = (this.MessageWriterSettings.LibraryCompatibility == ODataLibraryCompatibility.UseLegacyVariableCasing) ? EdmLibraryCompatibility.UseLegacyVariableCasing : 0

```

Comment:
Don't have `UseLegacyVariableCasing` as a property of `CsdlXmlWriterSettings`. Instead, the property should be of type `EdmLibraryCompatibility` in order to accommodate multiple compatibility flags combined together.

Reply:
Question: Should we have one flag for ScaleVariable and other flag for SridVariable?
Personally I think a single flag could be used to control both.
In which case `UseLegacyVariableCasing` might be a better fit...

Reply:
Having one flag is enough.

Reply:
The property is of that type. It's the name that needs to be changed.